src/main/java/trackmodel/TrackModel.java

```diff
@@ -330,11 +332,25 @@ protected void clearDB() {
      *
      * @param      blockId  The block identifier
      *
-     * @return     True if occupied, False otherwise.
+     * @return     True if occupied (or broken), False otherwise.
      */
     public boolean isOccupied(int blockId) {
         this.update();
 
+        switch (this.getStatus(blockId)) {

```

Comment:
Did you forget that Profetta told us that WC/TkM do not see the repair status of the track? If anything is wrong with it, it is considered occupied.

Reply:
Thats what this code is doing...

Reply:
getStatus is still a public method, yes?

Reply:
The track model does have to keep track of when a block in being repaired. The CTC still has to notify the track that it is being repaired, it just can read the track status. 
GetStatus is public, but probably should be protected. 
SetStarus needs to remain public. 
All I changed here is to have the track model actually react to the status of the block

Reply:
Can't read the track status*

Reply:
Yeah, and I'm pretty sure Profetta would tell you that even using the class BlockStatus, even without any external interface to access it (which you do still have), he would say that means you know more than you should know. He talked about having a dependency on BlockStatus in subsystem review.

Reply:
OK, this isn't the PR to block for this, for sure, but I think we need to get rid of `getStatus` at least at some point.

Reply:
I agree with that. I'll create an issue and be sure to add it to my next PR

Reply:
#83